### PR TITLE
porting tacotron2 from nvidia's repo

### PR DESCRIPTION
**Pull request by yangarbiter**
[Original implementation](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/SpeechSynthesis/Tacotron2)

Difference in training loop:
- Use pytorch's DistributedDataParallel instead of the one in apex.
- Drop amp mix integer training since it doesn't give much speed boost (according to their README).
- Load dataset from torchaudio.datasets instead of downloading it separately.
  - The data preprocessing is also rewritten with torchaudio's transforms.
- Add tensorboard for visualization.
- Drop their DLLogger and use python's logging utility.
- Simplifies the text preprocessing utility.

### CODE REVIEW

**Comment by vincentqb** [comment]
how does the model implementation compare to https://github.com/pytorch/audio/pull/855 ?

**Review by vincentqb — COMMENTED** [review]
please break down the PR into many separate PRs.

- one for the model and its tests (the model should be added with an underscore prefix since it has not been fully vetted yet)
- one for the text utilities
- one for the loss function
- one for the training
- one for inference (all the previous code should work without this one)

**Comment by vincentqb** [comment]
please keep in mind that we aim at following a different convention for the training loop, see [internal](https://docs.google.com/document/d/1gap2DWVKHCo_my8cRxQFiHgfz4-JzGO8nsr1XYHFz1o/edit#)
